### DISCUSSION TITLE
⚠️ Important Update: Deprecation of HTTP-API Versions

### DISCUSSION THREAD

**Discussion by FlorianKoerner** [dicebear/dicebear#374] [Announcements]
I am want to inform you of some upcoming changes to the DiceBear platform. In an effort to provide you with the most secure and efficient service possible, I have decided to sunset older versions of the HTTP-API, specifically versions `4.x`, `4.4` - `4.10` and `5.0` - `5.3`. These versions will no longer be available after **September 29, 2023**.

I understand that this may come as sudden news (but was [communicated](https://github.com/orgs/dicebear/discussions/255#user-content-new-api-endpoint)) and may cause some concern. Please understand that this decision is based on platform security considerations. Continuing to run software on outdated dependencies can pose security risks, and it is my responsibility to prevent any potential issues.

Regarding access to the APIs, please note that they will only be accessible via `api.dicebear.com`. Although this change has been communicated since the launch of version 5, it's important to reiterate it during this transition period.

~To ensure that your applications continue to work visually, all incoming requests to the now unavailable APIs will be returned with a default image.~

> Edit: 2023-12-04: The fallback avatar for the old API endpoint eats up a lot of money. That's why it is now deactivated.

As for alternatives, I strongly recommend upgrading to versions `5.x`, `6.x`, or `7.x`, which are still supported by the HTTP-API. Alternatively, you could choose to [host the HTTP-API yourself](https://www.dicebear.com/guides/host-the-http-api-yourself). I have set up a [Docker Hub repository](https://hub.docker.com/r/dicebear/api) for this purpose, where [tag 1.0](https://hub.docker.com/layers/dicebear/api/1.0.0/images/sha256-25cf53934e2671dd6900ded8dc0a7b87aa22b21373b25a0dd5b5cba87cdd2a99?context=explore) continues to support discontinued versions at your own risk. Alternatively, you can use version 4 of the JS-Library.

For more information about the API and these changes, please see the [documentation](https://www.dicebear.com/how-to-use/http-api).

I understand the concerns that such changes may cause and appreciate your understanding and flexibility.

**Comment by eli-s-r** [comment]
How do we retain the same seed when updating to the new API endpoint? For example, if we were previously using `https://avatars.dicebear.com/api/bottts/xyz.svg`, what new endpoint do we use to obtain the same avatar? It seems that `https://api.dicebear.com/7.x/bottts/svg?seed=xyz` doesn't return the same one.

**Reply by FlorianKoerner** [reply]
Hi @eli-s-r 

Between major versions, there may be differences between the generated avatars. This is due to the following reasons:

- The avatar style has been optimized, which leads to different calculations from the PRNG.
- New features have been added that also want to be taken into account, which changes the result.
- Bugfixes that have a direct impact on the output, such as https://github.com/dicebear/dicebear/issues/274.

Guaranteed the same avatar with the same settings you get only under the same major version. This means that you must continue to use version 4 if you want the same result as before. I mentioned the possibilities for this in the first post.

**Reply by FlorianKoerner** [reply]
And as the documentation states, do not use the HTTP API instance we provide for free for commercial purposes. https://www.dicebear.com/how-to-use/http-api/#fair-use

The alternatives here remain the same:

1. host your own instance of the HTTP API.
2. include the JS library directly on the page.
    - https://www.npmjs.com/package/@dicebear/avatars
    - https://www.npmjs.com/package/@dicebear/avatars-bottts-sprites

**Reply by eli-s-r** [reply]
Thanks for the detailed info @FlorianKoerner!
